python/cudf/cudf/core/column/column.py

```diff
@@ -1281,7 +1281,7 @@ def column_empty(
         data = None
         children = (
             build_column(
-                data=as_device_buffer_like(
+                data=as_buffer(
```

Comment:
General question: normally, in cases where we _know_ that we're passing an argument to `as_buffer` that is not a `Buffer` (aka where we are explicitly constructing an `rmm.DeviceBuffer`) I would recommend that we should use that knowledge and just call the `Buffer` constructor directly. However, I assume that in the next step we're going to use this function to also dispatch to `SpillableBuffer` vs `Buffer` right? If so, I wonder if we should protect ourselves further against instantiating the wrong object by maybe disabling the constructor and instead implementing a factory. What do you think?

Reply:
CC @shwina 
The case that I'm concerned about is someone manually doing `Buffer(...)` somewhere that should use `as_buffer` in a `SpillableBuffer`-enabled world.

Reply:
>  disabling the constructor and instead implementing a factory.

Agree

Reply:
> If so, I wonder if we should protect ourselves further against instantiating the wrong object by maybe disabling the constructor and instead implementing a factory. What do you think?

I think this would be great, cuML use to call  `Buffer(...)` directly but not anymore AFAICS.

Reply:
Do we need deprecation warning?

Reply:
This is done now, which also makes this PR `breaking`.

Reply:
https://github.com/rapidsai/cudf/pull/12009#discussion_r1009694262

Reply:
This is the main outstanding item left for me, then I think this PR will get a :heavy_check_mark: from me.